src/libraries/System.Private.CoreLib/src/System/Buffers/Text/Ascii.cs

```diff
@@ -0,0 +1,1244 @@
+        /// <remarks>
+        /// The contents of <paramref name="value"/> are modified in-place. Lowercase
+        /// ASCII <see cref="byte"/>s are converted to uppercase. Non-ASCII <see cref="byte"/>s
+        /// remain unchanged.
+        /// </remarks>
+        public static void ToUpperInPlace(Span<byte> value)
+        {
+            foreach (ref byte refElement in value)
+            {
+                refElement = ToUpper(refElement);
+            }
+        }
+
+        /// <summary>
+        /// Performs an in-place uppercase ASCII conversion of a buffer's contents.
+        /// </summary>
+        /// <param name="value">The buffer over which to operate.</param>
+        /// <remarks>
+        /// The contents of <paramref name="value"/> are modified in-place. Lowercase
+        /// ASCII <see cref="char"/>s are converted to uppercase. Non-ASCII <see cref="char"/>s
+        /// remain unchanged.
+        /// </remarks>
+        public static void ToUpperInPlace(Span<char> value)
+        {
+            foreach (ref char refElement in value)
+            {
+                refElement = ToUpper(refElement);
+            }
+        }
+
+        /*
+         * Performs case conversion on a single value, converting [A-Z] -> [a-z] or vice versa.
+         * All values outside [A-Za-z] - including non-ASCII values - are unchanged.
+         */
+
+        /// <summary>
+        /// Converts an ASCII <see cref="byte"/> to lowercase.
+        /// </summary>
+        /// <param name="value">The <see cref="byte"/> to convert.</param>
+        /// <returns>The lowercase form of <paramref name="value"/> if <paramref name="value"/>
+        /// is an uppercase ASCII <see cref="byte"/>; otherwise, returns <paramref name="value"/> unchanged
+        /// if <paramref name="value"/> is already lowercase or is not an ASCII <see cref="byte"/>.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static byte ToLower(byte value)
+        {
+            if (UnicodeUtility.IsInRangeInclusive(value, 'A', 'Z'))
+            {
+                value = (byte)(value | 0x20u);
+            }
+            return value;
+        }
+
+        /// <summary>
+        /// Converts an ASCII <see cref="char"/> to lowercase.
+        /// </summary>
+        /// <param name="value">The <see cref="char"/> to convert.</param>
+        /// <returns>The lowercase form of <paramref name="value"/> if <paramref name="value"/>
+        /// is an uppercase ASCII <see cref="char"/>; otherwise, returns <paramref name="value"/> unchanged
+        /// if <paramref name="value"/> is already lowercase or is not an ASCII <see cref="char"/>.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static char ToLower(char value)
+        {
+            if (UnicodeUtility.IsInRangeInclusive(value, 'A', 'Z'))
+            {
+                value = (char)(byte)(value | 0x20u);
```

Comment:
Add the comment why the cast to `byte` is beneficial? It makes sense, but it's not obvious.
I mean `// on x86, extending BYTE -> DWORD is more efficient than WORD -> DWORD`